dashboard/test/controllers/policy_compliance_controller_test.rb

```diff
@@ -78,6 +78,19 @@ class PolicyComplianceControllerTest < ActionDispatch::IntegrationTest
     end
   end
 
+  test "if a user enters their own email as their parent email, should just redirect back" do
+    user = create(:young_student, :without_parent_permission, email: 'test@studentemail.com')
+    sign_in user
+
+    assert_emails 0 do
+      post '/policy_compliance/child_account_consent', params:
+        {
+          'parent-email': 'test@studentemail.com',
```

Comment:
Sorry I didn't think about this until now, but children could easily work around this validation by using the "plus addressing", for example: test+parent@studentemail.com. Maybe some code like this would work:
```
def strip_plus_addressing(email)
  email.sub(/\+[^@]+@/, '@')
end

email = "john.doe+newsletter@example.com"
stripped_email = strip_plus_addressing(email)
puts stripped_email  # => "john.doe@example.com"
```

You could optionally add this to the front end validation, but I think the backend is the only required one.

Reply:
Done! Merging when CI passes.